### PR TITLE
Windows support: fix AtomicWriteFile function

### LINKED ISSUE #2616: Windows support: fix AtomicWriteFile function
The `disk` agent keymanager plugin fails with the following error when running on Windows:
```
unable to generate private key: rpc error: code = Internal desc = keymanager(disk): failed to generate key: unable to write entries: sync .data: The handle is invalid.
```
The reason behind this is that the call [dir.Sync()](https://github.com/spiffe/spire/blob/main/pkg/common/diskutil/file.go#L27) ends up in a call to the [FlushFileBuffers](https://docs.microsoft.com/en-gb/windows/win32/api/fileapi/nf-fileapi-flushfilebuffers) function, which does not expect a handle of a directory, and only a file handle is expected.

### PR DESCRIPTION
**Pull request by MarcosDY**
Windows does not allows using file.Sync to folders, that makes our current code for diskutil to fails
To solve this add a new set of functions that will be used on windows to use a custom AtomicRename that uses `MOVEFILE_REPLACE_EXISTING|MOVEFILE_WRITE_THROUGH` which waits until everything is synced to disk.

It is tested in a windows 10.

**Which issue this PR fixes**
fixes #2616

### CODE REVIEW

File: pkg/common/diskutil/file.go [hunk 1]
```diff
@@ -1,3 +1,6 @@
+//go:build !windows
```
Comment (azdagron) [inline pkg/common/diskutil/file.go:1]: Maybe this file can be renamed `file_posix.go`?

File: pkg/common/diskutil/file_test.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+//go:build !windows
```
Comment (azdagron) [inline pkg/common/diskutil/file_test.go:1]: Is file_windows_test.go a duplicate of this file? If so, would it be better to remove the build constraints from this file?

Comment (azdagron) [reply pkg/common/diskutil/file_test.go:1]: I see now that there are mode differences... hmm, i'm ok either way but one alternative is to detect the os at runtime and change the expected mode accordingly.

**Review by azdagron — COMMENTED** [review]
Thanks, @MarcosDY ! Just a few comments.

File: pkg/common/diskutil/file_windows.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	to, err := syscall.UTF16PtrFromString(newPath)
+	if err != nil {
+		return err
+	}
+
+	return moveFileEx(from, to, movefileReplaceExisting|movefileWriteThrough)
+}
+
+func moveFileEx(from *uint16, to *uint16, flags uint32) (err error) {
```
Comment (amartinezfayo) [inline pkg/common/diskutil/file_windows.go:79]: Is this really needed? Can't we just import `"golang.org/x/sys/windows"` and call `windows.MoveFileEx` instead?

**Review by amartinezfayo — COMMENTED** [review]
Looking great!
I have one comment about the `moveFileEx` function. I think that we may just use `windows.MoveFileEx`.

**Review by amartinezfayo — COMMENTED** [review]
One final comment from my side.

**Review by amartinezfayo — APPROVED** [review]
Thanks @MarcosDY, LGTM!

**Pull request merged**